Ui/fix brackdrop

Luminilion opened the pull request:
Cleaned Backdrop (BottomSheet) UI & refactored design of contained elements

Commit:
Enforced ButterKnife & reused Fragments

Commit:
Enforced ButterKnife

Commit:
Removed unused imports

Commit:
Set collapsed state of bottom sheet when returning

Commit:
Created new drawables to use in App UI

Include :
- rectangle slight rounded transparent background shape
- rectangle slight rounded primaryColor background shape
- rectangle top&right visible slides only transparent background
- rectangle top-only visible slides only transparent background

Commit:
Refactored unjoined event list item design

Commit:
Refactored bottomSheet xml

Commit:
Adapted IDs for coherence

Commit:
Added light_gray, more dimensions + bottomSheet&borderlessButton styles

Commit:
Added interactive arrow on Backdrop

Commit:
Refactored UI for text fit, layouts and sizes

Commit:
Merge branches 'master' and 'ui/Fix-brackdrop' of github.com:Susanfe/sdp-event-management into ui/Fix-brackdrop

# Conflicts:
#	app/src/main/java/ch/epfl/sweng/eventmanager/ui/event/interaction/EventShowcaseActivity.java
#	app/src/main/res/values/dimens.xml

Commit:
Modifications on scopes (cf merge errors & warnings)

vquelque commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/event/interaction/EventShowcaseActivity.java:
```diff
@@ -48,9 +48,9 @@
-    private Fragment scheduleParentFragment;
+    public Fragment eventMapFragment;
+    public Fragment newsFragment;
+    public Fragment scheduleParentFragment;
```
I think this should remain private

Luminilion replied:
If they are private, they cannot be used by other activities such as EventMainFragment to recycle the created fragments (if any). What I wanted to do was reusing the already created fragments instead of recreating them every time.

vquelque commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/event/interaction/fragments/EventMainFragment.java:
```diff
@@ -84,20 +88,27 @@ public void onResume() {
         return view;
 
     }
+
```
Better make a public changeFragment() method in the showcase activity as we don't have to have these weird public fragments

Luminilion replied:
Yes I got your point ok

Commit:
Implemented bottomsheet behavior + strings styles dimen update

Commit:
Modified drawables

- color : all -> Primary
- radius : var name change
- added person drawable for user management

Commit:
refactored event picking layout

Commit:
Var name change

Commit:
Set transparency on bottom sheet + refactored CardViews for simpler ones

Commit:
Refactored bottom sheet item button style

Added colors + button drawable + style
Deleted old style

Luminilion committed:
Merge branch 'master' into ui/Fix-brackdrop

Commit:
Fixed bug for API <21 and refactored use of changeFragment() for reuse

Commit:
Merge remote-tracking branch 'origin/ui/Fix-brackdrop' into ui/Fix-brackdrop
